src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/NetworkManager.cs

```diff
@@ -2,30 +2,30 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.IO;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.SpectatorView
 {
     public abstract class NetworkManager<TService> : CommandRegistry<TService>, INetworkManager where TService : Singleton<TService>
     {
-        private float lastReceivedUpdate;
-
+        [Tooltip("Default prefab for creating an INetworkConnectionManager.")]
         [SerializeField]
-        protected TCPConnectionManager connectionManager = null;
+        private GameObject defaultConnectionManagerPrefab = null;
+        private GameObject connectionManagerGameObject = null;
+        protected INetworkConnectionManager connectionManager = null;
 
-        public bool AttemptReconnectWhenClient
-        {
-            get { return attemptReconnectWhenClient; }
-            set { attemptReconnectWhenClient = value; }
-        }
-        private bool attemptReconnectWhenClient = true;
+        private float lastReceivedUpdate;
+        private INetworkConnection currentConnection;
 
-        private SocketEndpoint currentConnection;
+        private static readonly List<INetworkConnection> emptyConnectionList = new List<INetworkConnection>();

```

Comment:
>private static readonly List<INetworkConnection> emptyConnectionList = new List<INetworkConnection>(); [](start = 8, length = 102)

This could be "private static readonly IReadOnlyList<INetworkConnection> emptyConnectionList = Array.Empty<INetworkConnection>();"  An empty array implements IReadOnlyList and requires less space in memory (and is truly read-only).

Reply:
oh cool, this is exactly what i was looking for :)